### PR TITLE
fix: oci pull directory conflict when download twice

### PR DESCRIPTION
**Pull request by yxxhero**
see: https://github.com/helmfile/helmfile/issues/910

### CODE REVIEW

File: pkg/state/state.go [hunk 1]
```diff
@@ -3461,15 +3460,19 @@ func (st *HelmState) getOCIChart(release *ReleaseSpec, tempDir string, helm helm
...
+	chartPath := filepath.Join(pathElems...)
 
-	err := helm.ChartPull(qualifiedChartName, chartPath)
-	if err != nil {
-		return nil, err
-	}
-	err = helm.ChartExport(qualifiedChartName, chartPath)
-	if err != nil {
-		return nil, err
+	if st.fs.DirectoryExistsAt(chartPath) {
```
Comment (xiaomudk) [inline pkg/state/state.go:3465]: In the past, we used to pull OCI charts every time to ensure they were up to date. However, the current approach of only checking for directory existence may lead to the inability to fetch the latest OCI charts. In extreme cases, even if the directory exists, but is empty, the OCI charts will not be downloaded.

Comment (yxxhero) [reply pkg/state/state.go:3465]: make sense.

Comment (yxxhero) [reply pkg/state/state.go:3465]: one question: it will error when fetch twice. so OCI chart will not be updated now.

**Comment by yxxhero** [comment]
@mumoshu ping

File: pkg/state/state.go (hunk 1 above)
Comment (mumoshu) [reply pkg/state/state.go:3465]: I believe this is the best approach today...
I hope helm-export is enhanced to somehow export it only when the chart has been updated remotely (e.g. by checking the hashsum of the previously downloaded and exported chart against the new hashsum, or something like that)
And I believe doing that in Helmfile might be inefficient and not practical.

**Review by mumoshu — APPROVED** [review]
LGTM. Thank you so much for your continued effort @yxxhero!

**Pull request merged**